src/include/serial.h

```diff
@@ -0,0 +1,16 @@
+extern unsigned int serial_enabled;
+extern unsigned int serial_initialized;
+
+#ifdef __cplusplus
+#define OPT_EXTERN extern "C"
+#else
+#define OPT_EXTERN
+#endif
+
```

Comment:
we should probably move this definition to some common header like stddef.h at some point, probably going to be using it a lot.

Reply:
Yeah, I was wondering a better place for it to live assuming this was acceptable.